webdriver/support/fixtures.py

```diff
@@ -142,3 +144,41 @@ def inner(path, query="", fragment=""):
                                   fragment))
         return rv
     return inner
+
+def create_dialog(session):
+    """Create a dialog (one of "alert", "prompt", or "confirm") and provide a
+    function to validate that the dialog has been "handled" (either accepted or
+    dismissed) by returning some value."""
+
+    def create_dialog(dialog_type, text=None, result_var=None):
+        assert dialog_type in ("alert", "confirm", "prompt"), (
+               "Invalid dialog type: '%s'" % dialog_type)
+
+        if text == None:
+            text = ""
+
+        assert isinstance(text, basestring), "`text` parameter must be a string"
+
+        if result_var == None:
+            result_var = "__WEBDRIVER"
+
+        assert re.search(r"^[_$a-z$][_$a-z0-9]*$", result_var, re.IGNORECASE)
+
+        # Script completion and modal summoning are scheduled on two separate
+        # turns of the event loop to ensure that both occur regardless of how
+        # the user agent manages script execution.
+        spawn = """
+            var done = arguments[0];
+            setTimeout(function() {{
+                done();
+            }}, 0);
```

Comment:
Yup. [That's the escape sequences for literal braces in Python's
`string.format` syntax](https://docs.python.org/2/library/string.html):

> Format strings contain “replacement fields” surrounded by curly braces {}.
> Anything that is not contained in braces is considered literal text, which is
> copied unchanged to the output. If you need to include a brace character in
> the literal text, it can be escaped by doubling: {{ and }}.